### PR TITLE
Lich King's Mind Control

### PR DESCRIPTION
**Pull request by zumbak04**
## Changelog:
- Lich King's vassals are less likely to revolt if has high Dark Authority and more likely if it has low.
- Having high Dark Authority, undead vassals are less likely to refuse Lich King's interactions.

## Tests:
- [x] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [x] The mod takes less than 5.5 GB in the Task Manager (Windows)
- [x] Revert tests

# How to test:
Check these features. Remember, you can lower and raise Dark Frenzy with the Debug Mode enabled by clicking the Lich King and looking under the **_DEBUG Interactions_**.

So you may raise/low Dark Frenzy and see how the AI reacts.

### CODE REVIEW

**Review by Remerod — COMMENTED** [review]
I tested 2 games, one with High Dark Authority and one with 0 

<details>
<summary>Click to expand</summary>

I tested first with High Autority on the bookmark 605 ->
_Having high dark authority seems to work_
image


image
_at the year 616, 8 after start Dar'khan independence faction demands independence and shortly after that a frost vrykul/iskronir faction follows_

image
_third independence faction_

image
_4th independence war at 638, although this is a populist uprising_

image
_5th independence war at 640_

image
_6th independence war, at 652_

**Summary of High Authority:** 
during the 53 years i tested i had 6 independence revolts.

So started my test with 0 dark authority on the bookmark 605

image
_So far I have had a holy uprising 10 years into the game_

image
_but shortly after i had my first independence war at 616_

image
_Second independence war at 627_

image
_Third Independence war at 651 AD_

**Summary on low authority test:**
During the 50 years i played with 0 dark authority I had 3 independence wars

**Whole test Summary**: 
so during both test runs i mainly managed dark frenzy with cheats and won war with cheats other then that i did nothing really. The High Dark Authority test i did my vassals was a little bit more successful. but i had twice the amount of independence wars. The Second test i did i made sure i had 0 dark frenzy at all times, and my realm was... well quite peaceful, some more raiders though and 3 independence wars  

</details>

**Comment by zumbak04** [comment]
> so during both test runs i mainly managed dark frenzy with cheats and won war with cheats other then that i did nothing really. The High Dark Authority test i did my vassals was a little bit more successful. but i had twice the amount of independence wars.

I can't judge based on such long runs because there may be other reasons. Anyway, I did my tests and made these modifiers a bit more fatal in 6a122c6362fc493e76f50f86bddc34f131a35c0e.

Also, for tests, I disabled one limitation: there are no factions the first 10 years of the game in 8efd800b0dffaccc03b379e72a3522df154137bd.

**Review by Metsmorphosis — APPROVED** [review]
first, run I did 10 years high dark authority going for Diplo focus used cheats to keep authority maxed and accepted all willing to become undead vassals I made Dar'Khan "king" so he dropped the faction then swayed others out of faction when another popped up I swayed and gifted to keep them out of factions and transferred vassals when necessary. if you aren't dumb or horribly unlucky you won't have to deal with independence factions Basically
High Dark Frenzy
mostly populist factions nothing more than a nuisance 

 the low dark frenzy run I did a little longer to wait out another independence faction because I did the same thing vassal and Dar'Khan wise and that ended up getting a war almost immediately in the first year or two used cheats to win because it was my 4.4k troops vs 122k troops but after i "won" I gifted all the people gold without cheats and swayed when I couldn't gift then waited 2 or so years for another faction to form. I'm thinking if you manage to win the first war you can prevent the second just because of the limitations of being at +100 or close to +100 opinion effects on joining a faction. all in all, things seem to be working as intended.

**Review by Remerod — COMMENTED** [review]
So tried a test run again 

<details>
<summary>Click to expand</summary>

So I started at the fronzen throne bookmark and I was just gonna wait 10 years without caring about dark frenzy to much, since you said factions won't rebel against me but here at 611 AD they do. 

image

image

but I'll finish this rebellion and and wait to 615 and do a save there before going low vs high dark influence

**High Dark Authority** 
having almost 100 dark authority at 615 and onward most factions kept far under 18% of strength, but most of my dangerous vassals was not gonna rebel against me intill 621 so i waited for that, and almost immediately a independence faction formed 

image

Although their power diminished quickly and did not rebel after their initial power boost when formed.

image

So played to 630, I never had an idenepandance faction after the initial 10 years before i made my save and made sure i had high to full dark authority, no more then 2-3 supported the faction and the biggest negative modifers on them would come from: wanting a seat on my council, Unholy Congrestion and when i havn't been to war war for a while it could go up to - 30 (at peace).

**Low Dark Authority** 
So waiting again for my formally imprisoned vassals to be able to rebel again i had a look at faction from 615 to 621 and most kept under 10% with one exception of the Fjortir Mord'truari populist who would shift from 30-50% military power with 11 members supporting it, other would be most 1-2 people supporting it.

image

A little under one year after the strong vassals could rebel again they made an independence faction of 214% Military Power with 17 members in it.

image

forgot to mention the first one but during my low dark authority run i have now had 2 peasant rabble revolts (i kind of just wait of the first one) this faction have 11% military power with 6 members supporting it 

image

After playing to 630 i can see that during the 15 years i played after my 10 year save with 0 Dark Authority I had 1 big Independent War, with a lot more character involved in it, i would probably have another one shortly after 630 when they can rebel again. beside that i had two peasant rabbles wanting me to be more lenient with taxes (lol should this be a thing?)

**Summary**
Clearly this did not work, since i had a independant faction 6 years after game start.
> Also, for tests, I disabled one limitation: there are no factions the first 10 years of the game in 8efd800.

But after 10 years and i made my save and have 2 test run from that save with one almost always full at 100 and one at 0 i could really see the difference in power and people joining the independence factions. On the other minor populist faction i saw almost no change, most was under 10% at most times and rarely went up to 20% in military strength, with one exception on the 0 dark authority that went all up to 56% percent at one time but was mostly around 30% and never rebelled. So I would say that there was a great difference in independence faction strength between 100 and 0 dark authority 

</details>

**Comment by zumbak04** [comment]
> you said factions won't rebel against me but here at 611 AD they do.

Otherwise, it was a limitation and I disabled it.

> On the other minor populist faction i saw almost no change, most was under 10% at most times and rarely went up to 20% in military strength, with one exception on the 0 dark authority that went all up to 56% percent at one time but was mostly around 30% and never rebelled.

A populist faction? A county of different faith or culture may create it and join it. If it's of different culture, it's not an undead county thus it's not affected by this feature.

**Review by Remerod — COMMENTED** [review]
well, I think this seems to be working as intended. Approved

File: localization/english/wc_character_interactions_l_english.yml [hunk 1]
```diff
@@ -9,4 +9,6 @@
  nothing:0 "Nothing"
 
  TARGET_SCOURGE_JAILOR_REASON:1 "[target.GetShortUIName|U] is a [scourge_jailor|E]: $VALUE|=+0$"
- RECIPIENT_SCOURGE_JAILOR_REASON:1 "[recipient.GetShortUIName|U] is a [scourge_jailor|E]: $VALUE|=+0$"
\ No newline at end of file
+ RECIPIENT_SCOURGE_JAILOR_REASON:1 "[recipient.GetShortUIName|U] is a [scourge_jailor|E]: $VALUE|=+0$"
+ 
+ LICH_KING_VOICE:0 "Hears your voice in the head: $VALUE|=+0$"
```
Comment (MemeWizard) [inline localization/english/wc_character_interactions_l_english.yml:14]: probably the lich king trait should be linked here for greater clarity

Comment (zumbak04) [reply localization/english/wc_character_interactions_l_english.yml:14]: > 

How?

Comment (MemeWizard) [reply localization/english/wc_character_interactions_l_english.yml:14]: Hears [lich king(trait)] voice in the head ?

**Review by Remerod — APPROVED** [review]
Sorry made a comment instead of an Approve, this time it's real! :P

File: localization/english/wc_character_interactions_l_english.yml (hunk 1 above)
Comment (zumbak04) [reply localization/english/wc_character_interactions_l_english.yml]: Done in d801618c1fc4203afa18b879429c4f407421f56f.

**Pull request merged**